### PR TITLE
enable dupe run protection using...

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

enable dupe run protection using fkirc/skip-duplicate-actions; don't allow concurrent runs or re-running the same release (version + phases)

Change-Id: Id3dfd0eb2b84e4cbfe64c02f00f31cb992877a73
Signed-off-by: nickboldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
N/A (or see commit message above for issue number)

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: .github/workflows/release-orchestrate-overall.yml [hunk 1]
```diff
@@ -12,7 +12,22 @@ on:
...
         default: '1,2,3,4,5,6,7'
 jobs:
+  pre_job:
+    runs-on: ubuntu-latest
+    # Map a step output to a job output
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@master
```
Comment (nickboldt) [inline .github/workflows/release-orchestrate-overall.yml:22]: should use a SHA instead of branch to avoid potential for malicious code

Comment (mkuznyetsov) [reply .github/workflows/release-orchestrate-overall.yml]: added

**Comment by mkuznyetsov** [comment]
I pushed some commits, and ensured that this works

**Review by mkuznyetsov — APPROVED** [review]

**Comment by nickboldt** [comment]
Excellent. Now we just need to talk to emo@eclipse.org to find out what the policies are around using un-CQ'd 3rd party code in RUNS WITH build time dependencies :)

**Comment by nickboldt** [comment]
EMO @ Eclipse has confirmed that we can use 3rd party GH actions as long as they don't modify Eclipse project source code. \o/

**Pull request merged**